linetools/lists/linelist.py

```diff
@@ -0,0 +1,236 @@
+"""
+Module for LineList Class
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+import numpy as np
+import os
+
+from astropy import units as u
+from astropy.units.quantity import Quantity
+from astropy import constants as const
+from astropy.io import fits
+from astropy.table import QTable, Table, vstack
+
+#from xastropy.xutils import xdebug as xdb
+
+from linetools.lists import parse as lilp
+
+#
+class LineList(object):
+    '''Class to over-load Spectrum1D for new functionality not yet in specutils
+
+    Parameters:
+    ----------
+    llst_keys: str or list  
+      Input to grab line list.  Current options are:
+       'ISM'     :: "All" ISM lines
+       'Strong'  :: Strong ISM lines
+       'HI'      :: HI Lyman series
+       'H2'      :: H2 (Lyman-Werner)
+       'CO'      :: CO UV band-heads [not yet implemented]
+       'EUV'     :: Key EUV lines (for CASBAH project)
+       'Gal_E'   :: Galaxy emission lines (HII)
+       'Gal_A'   :: Galaxy absorption lines (stellar)
+       'AGN'     :: Key AGN lines
+    gd_lines: list, optional
+      List of wrest for lines to use (drawn from input linelist)
+    verbose: bool, optional
+      Give info galore if True
+    '''
+    # Init
+    def __init__(self, llst_keys, gd_lines=None, verbose=True):
+
+        # Error catching
+        if type(llst_keys) not in [str,list,unicode]:
```

Comment:
Use `isinstance(llst_keys, (type1, type2))` instead of `type(llst_keys) in (type1, type2)`.  It handles subclasses, unlike `type`.

Also, use `isinstance(val, basestring)` instead of  `isinstance(val, (str, unicode))`.